packages/arb-token-bridge-ui/src/components/TransferPanel/TokenApprovalDialog.tsx

```diff
@@ -95,8 +95,12 @@ export function TokenApprovalDialog(props: TokenApprovalDialogProps) {
           gasEstimate = constants.Zero
         } else {
           const cctpTransferStarter = new CctpTransferStarter({
-            sourceChainProvider: isDepositMode ? l1.provider : l2.provider,
-            destinationChainProvider: isDepositMode ? l2.provider : l1.provider
+            sourceChainProvider: isDepositMode
+              ? parentChainProvider
+              : childChainProvider,
+            destinationChainProvider: isDepositMode
+              ? childChainProvider
+              : parentChainProvider
```

Comment:
to be tested, but know that we have `useNetworks` hook, we can use `sourceChainProvider` here

Reply:
Yeah, makes sense. Updated.